File: android/src/main/java/com/mcodex/nitroarchive/application/ports/ArchiveInput.kt
package com.mcodex.nitroarchive.application.ports

interface ArchiveInput {
    suspend fun openForRandomAccess(): ByteArray
    val estimatedSize: Long?
}

interface ArchiveEngineSession {
    fun entry(at: String): EntryDescriptor?
    val entries: List<EntryDescriptor>
    val archiveData: ByteArray?
}

File: ios/Application/Ports/ArchiveInput.swift
import Foundation

protocol ArchiveInput {
  func openForRandomAccess() async throws -> Data
  var estimatedSize: UInt64? { get }
}

protocol ArchiveEngineSession {
  func entry(at path: String) -> EntryDescriptor?
  var entries: [EntryDescriptor] { get }
  var archiveData: Data? { get }
  var archiveURL: URL? { get }
}
